README.md
Description: panel representing vlvt io (Mkssetpoint server)

assigned to: Giacomo Strangolino

hw referent: Graziano Scalamera

build 

qmake && make

giacomo.strangolino@elettra.trieste.it




CHANGELOG


release_04:	porting to qtango6 (installed in /usr/local/qtango) and qt5


release_03:	porting to qtango6 (installed in /usr/local/qtango) and qt5
release_01: first release based on Luca Rumiz and Riccardo G. specifications


src/vlvt_io.h
/***************************************************************************
 *   Copyright (C) 2007 by Vincenzo Forchi`,,,   *
 *   vincenzo.forchi@elettra.trieste.it   *
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 *   This program is distributed in the hope that it will be useful,       *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
 *   GNU General Public License for more details.                          *
 *                                                                         *
 *   You should have received a copy of the GNU General Public License     *
 *   along with this program; if not, write to the                         *
 *   Free Software Foundation, Inc.,                                       *
 *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
 ***************************************************************************/


#ifndef vlvt_io_H
#define vlvt_io_H

#include "ui_vlvt_io.h"

class VlvtIo: public QWidget
{
    Q_OBJECT

public:
    VlvtIo(QWidget * =NULL);
    ~VlvtIo();

public slots:
    void on_tabWidget_currentChanged(int = 0);

protected:
    void showEvent(QShowEvent *);

private:
    Ui::VlvtIo ui;

};


#endif


src/vlvt_io.cpp
/***************************************************************************
 *   Copyright (C) 2007 by Vincenzo Forchi`,,,   *
 *   vincenzo.forchi@elettra.trieste.it   *
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 *   This program is distributed in the hope that it will be useful,       *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
 *   GNU General Public License for more details.                          *
 *                                                                         *
 *   You should have received a copy of the GNU General Public License     *
 *   along with this program; if not, write to the                         *
 *   Free Software Foundation, Inc.,                                       *
 *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
 ***************************************************************************/


#include "vlvt_io.h"
#include <qtango.h>
#include <elettracolors.h>

VlvtIo::VlvtIo(QWidget *parent) : QWidget(parent)
{
    /* Porting to QTango3 means setStateColor global through Config::instance() */
    Config::instance()->setStateColor(Tango::OPEN, EColor(Elettra::green));
    Config::instance()->setStateColor(Tango::CLOSE, EColor(Elettra::darkYellow));

    ui.setupUi(this);

    ui.tabWidget->widget(1)->layout()->setSpacing(3);
    ui.tabWidget->widget(1)->layout()->setMargin(4);
    on_tabWidget_currentChanged(0);
    /* Temporary hack to show a temporary vlvt_io state. Inside the .ui the temporary vlvt_io is configured
  * through the bit 2. If the panel was not launched with --temptest parameter, then remove the information
  * related to the bit number 2.
  */
    if(!qApp->arguments().contains("--temptest"))
    {
        ui.tVlvtIoStat->setBooleanDisplay(2, "---", "---", EColor(Elettra::white), EColor(Elettra::white));
    }
    if(qApp->arguments().contains("--fastvlvt_io"))
    {
	ui.tLabel_3->hide();
        ui.tVlvtIoStat->setBooleanDisplay(2, "---", "INRUSH", EColor(Elettra::white), EColor(Elettra::red));
        foreach(TPushButton *pb, findChildren<TPushButton *>())
            pb->setHidden(true);
    }
}

VlvtIo::~VlvtIo()
{
}    

void VlvtIo::showEvent(QShowEvent *e)
{
    QWidget::showEvent(e);
    on_tabWidget_currentChanged(ui.tabWidget->currentIndex());
}

void VlvtIo::on_tabWidget_currentChanged(int idx)
{
    resize(sizeHint());
}




src/main.cpp
/***************************************************************************
 *   Copyright (C) 2007 by Vincenzo Forchi`,,,   *
 *   vincenzo.forchi@elettra.trieste.it   *
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 *   This program is distributed in the hope that it will be useful,       *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
 *   GNU General Public License for more details.                          *
 *                                                                         *
 *   You should have received a copy of the GNU General Public License     *
 *   along with this program; if not, write to the                         *
 *   Free Software Foundation, Inc.,                                       *
 *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
 ***************************************************************************/


#include <TApplication>
#include "vlvt_io.h"
#include <tutil.h>
#include <X11/Xlib.h>
#include <QX11Info>
#include <QSplashScreen>

#define CVSVERSION "$Name:  $"
/* uncomment to set application color */
#include <elettracolors.h>

int main( int argc, char ** argv ) {

    TApplication a( argc, argv );
    /* setup some TApplication parameters */
    a.setPalette(EPalette("fermi"));
    a.setOrganizationName("Elettra");
    a.setApplicationName("VlvtIo");
    a.setProperty("author", "Giacomo");
    a.setProperty("hwReferent", "Graziano");
    QString version(CVSVERSION);
    a.setApplicationVersion(version);
    
    TUtil::instance()->setLoggingTarget(argv[0]);

    QPixmap pix(200,40);
    pix.fill(a.palette().color(QPalette::Window));
    QSplashScreen splash(pix);
    QFont f = splash.font();
    f.setPointSize(12);
    splash.setFont(f);
    splash.showMessage("Connecting to devices...", Qt::AlignCenter);
    splash.show();

    VlvtIo mw;
    if (argc > 1)
        mw.setWindowTitle(a.arguments()[1].split('/').last());
    else
        mw.setWindowTitle("VlvtIo");
    mw.show();

    splash.finish(&mw);

    /* register to window manager */
    Display *disp = QX11Info::display();
    Window root_win = (Window) mw.winId();
    XSetCommand(disp, root_win, argv, argc);
	
    return a.exec();
}
